--- a/net/minecraft/world/level/storage/loot/LootContextArg.java
+++ b/net/minecraft/world/level/storage/loot/LootContextArg.java
@@ -25,7 +25,7 @@
     }
 
     static <R> LootContextArg<R> of(final ContextKey<? extends R> contextParam) {
-        return () -> contextParam;
+        return (SimpleGetter)() -> contextParam;
     }
 
     static <R> Codec<LootContextArg<R>> createArgCodec(final UnaryOperator<LootContextArg.ArgCodecBuilder<R>> consumer) {
